### PR TITLE
docs: Add Conftest example

### PR DESCRIPTION
**Pull request by vdmgolub**

### CODE REVIEW

File: examples/conftest/README.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+Then use Conftest to test infrastructure cost changes against the policy.
+
+[//]: <> (BEGIN EXAMPLE)
+```yml
+pr:
+  - master
+
+variables:
+  API_KEY: $(apiKey)
+
```
Comment (hugorut) [inline examples/conftest/README.md:62]: FYI we don't need this anymore, you can simply remove this and reference `$(apiKey)` directly in the input. I'm updating other examples to remove this.

**Review by hugorut — APPROVED** [review]
LGTM - just remove unneeded variables declaration. I have also seen you tested the failure. 👍   :shipit:

**Comment by hugorut** [comment]
FYI - use the rebase and merge into the staging branch so we have a clean commit history when we eventually merge that one into master

File: .azure/pipelines/examples-test.yml [hunk 2]
```diff
@@ -3,6 +3,40 @@ pr:
...
+      - bash: >-
+          infracost breakdown --path=examples/conftest/code/plan.json
+          --format=json --out-file=/tmp/infracost.json
+        displayName: Run Infracost
+      - bash: >-
+          infracost output --path=/tmp/infracost.json
+          --format=azure-repos-comment --show-skipped
+          --out-file=/tmp/infracost_comment.md
+        displayName: Generate Infracost comment
+      - bash: conftest test --policy examples/conftest/policy /tmp/infracost.json
```
Comment (vdmgolub) [inline .azure/pipelines/examples-test.yml:38]: @hugorut Do you think we should keep the same approach by using `##vso[task.logissue type=error` output here?

Comment (hugorut) [reply .azure/pipelines/examples-test.yml:38]: hmm no - I prefer this approach. The `##vso[task.logissue type=error]` is a workaround as we need to fail the task output in a boolean way. I like that `conftest test` has this already built in.

**Pull request merged**